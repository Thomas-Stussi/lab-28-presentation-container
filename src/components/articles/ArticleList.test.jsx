import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

const testArray = [
  { title: 'title1', author: 'author1', description: 'description1' },
  { title: 'title2', author: 'author2', description: 'description2' },
  { title: 'title3', author: 'author3', description: 'description3' }
];

describe('ArticleList component', () => {
  afterEach(() => cleanup());
  it('render ArticleList component', () => {
    const { asFragment } = render(<ArticleList
      articles={testArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
