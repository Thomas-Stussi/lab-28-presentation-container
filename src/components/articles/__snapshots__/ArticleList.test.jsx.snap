// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component render ArticleList component 1`] = `
<DocumentFragment>
  <ul
    data-testid="articles"
  >
    <li>
      <h3>
        title1
      </h3>
      <p>
        By: author1
      </p>
      <p>
        Summary: description1
      </p>
    </li>
    <li>
      <h3>
        title2
      </h3>
      <p>
        By: author2
      </p>
      <p>
        Summary: description2
      </p>
    </li>
    <li>
      <h3>
        title3
      </h3>
      <p>
        By: author3
      </p>
      <p>
        Summary: description3
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
